File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_50830.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest50830(request: Request):
    user_id = request.query_params.get('id', '')
    parts = str(user_id).split(',')
    data = ','.join(parts)
    int(str(data))
    return {"updated": True}

File: Benchmarks/enterprise/java/jakarta/testcode/BenchmarkTest50830.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

@Path("/")
public class BenchmarkTest50830 {

    @GET
    @Path("/BenchmarkTest50830")
    @Produces(MediaType.APPLICATION_JSON)
    public Response BenchmarkTest50830(@HeaderParam("Host") String host, @Context HttpServletRequest request, @Context HttpServletResponse response) throws Exception {
        String hostValue = host != null ? host : "";
        String data;
        if (hostValue.length() > 256) { data = hostValue.substring(0, 256); }
        else { data = hostValue; }
        response.setHeader("X-Forwarded-For", data);
        return Response.ok("{\"ready\":true}", MediaType.APPLICATION_JSON).build();
    }
}
